### ISSUE TITLE
PHP 8: RestrictedFunction test failure

### ISSUE THREAD

**Issue by GaryJones** [Automattic/VIP-Coding-Standards#574]
## Bug Description

As seen, at https://travis-ci.org/github/Automattic/VIP-Coding-Standards/jobs/722253827, there is a test failure for PHP 8:

```sh
1) WordPressVIPMinimum\Tests\Functions\RestrictedFunctionsUnitTest::testSniff
[LINE 76] Expected 1 error(s) in RestrictedFunctionsUnitTest.inc but found 0 error(s).
```

It's related to [this line](https://github.com/Automattic/VIP-Coding-Standards/blob/271708fd64174e5c079ebc948991db355d88296d/WordPressVIPMinimum/Tests/Functions/RestrictedFunctionsUnitTest.inc#L76)

```php
\add_role(); // Error.
```

We have it marked down as [expecting 1 error](https://github.com/Automattic/VIP-Coding-Standards/blob/271708fd64174e5c079ebc948991db355d88296d/WordPressVIPMinimum/Tests/Functions/RestrictedFunctionsUnitTest.php#L47) on that line.

## Environment

Use `php -v` and `composer show` to get versions.

| Question                | Answer
| ------------------------| -------
| PHP version             | 8.0.0-dev (5th August 2020)
| PHP_CodeSniffer version | 3.5.5
| VIPCS version           | `develop`

**Comment by GaryJones** [comment]
Possibly a bug/change in the upstream abstract sniff, and [this `is_token_namespaced()` method](https://github.com/WordPress/WordPress-Coding-Standards/blob/1d0d10077fe9674c6903dbda886018f6ec7b4fce/WordPress/Sniff.php#L1462-L1497).

**Comment by jrfnl** [comment]
This is not directly related to the VIPCS repo and will be fixed when the PHPCS (3.x) tokenizer has been updated for PHP 8.
There are also numerous other sniffs - both in VIPCS as well as WPCS - which will be affected by this PHP change.

The underlying problem is this change in PHP: https://wiki.php.net/rfc/namespaced_names_as_token

This is the proposal for handling this in PHPCS: https://github.com/squizlabs/PHP_CodeSniffer/issues/3041

I intend for the PHPCSUtils abstracts will support this for older PHPCS versions (and for PHPCS 4.x) with a BC layer.

**Comment by jrfnl** [comment]
See: https://github.com/squizlabs/PHP_CodeSniffer/pull/3063

**Comment by GaryJones** [comment]
Since we can't fix this directly in VIPCS, short of requiring a future minimum version of PHPCS, I'm going to close this out.
